app/src/main/java/org/simple/clinic/searchresultsview/SearchResultsItemType.kt

```diff
@@ -15,6 +15,33 @@ sealed class SearchResultsItemType<T : ViewHolder>(adapterId: Long) : GroupieIte
   companion object {
     private const val HEADER_NOT_IN_CURRENT_FACILITY = 0L
     private const val HEADER_NO_PATIENTS_IN_CURRENT_FACILITY = 1L
+
+    fun generateListItems(
+        results: PatientSearchResults,
+        currentFacility: Facility
+    ): List<SearchResultsItemType<out ViewHolder>> {
+      if (results.visitedCurrentFacility.isEmpty() && results.notVisitedCurrentFacility.isEmpty()) return emptyList()
+
+      val itemsInCurrentFacility = if (results.visitedCurrentFacility.isNotEmpty()) {
```

Comment:
Break this function down to smaller ones? Improves readability. This is too much context to hold and one has to read the entire function to understand what is going on.

Reply:
Sure.